agoric/contract/src/kreadKit.js

```diff
@@ -1127,18 +1160,43 @@ export const prepareKreadKit = async (
             const royalty = multiplyBy(want.Price, royaltyRate);
             const platformFee = multiplyBy(want.Price, platformFeeRate);
 
-            itemsToSell.forEach((copyBagEntry) => {
+            const claimedIdAndRecorder = await Promise.all(
+              itemsToSell.map(async (copyBagEntry) => {
+                const [_, itemSupply] = copyBagEntry;
+                const array = Array.from(Array(Number(itemSupply)).keys());
+                const idAndRecorder = await Promise.all(
+                  array.map(async (i) => {
+                    const id =
+                      this.state.market.metrics.get('item').putForSaleAmount;
+                    marketFacet.updateMetrics('item', {
+                      putForSaleAmount: true,
+                    });
+                    const entryRecorder =
+                      await marketFacet.makeMarketItemRecorderKit(id);
+
+                    return [id, entryRecorder];
+                  }),
+                );
+                return idAndRecorder;
+              }),
+            );
+
+            let count = 0;
+            itemsToSell.forEach(async (copyBagEntry) => {
               const [itemObject, itemSupply] = copyBagEntry;
 
               for (let n = 0; n < itemSupply; n++) {
+                const [id, entryRecorder] = claimedIdAndRecorder.flat()[count];
```

Comment:
I'm confused why you need to flatten. You have nested loops over nested arrays.

Consider just iterating over the inner array in this inner loop